repo_health/index/static/client/src/components/base-component/base-stats.component.js

```diff
@@ -0,0 +1,47 @@
+/*
+* base-stats.component.js - (C) Copyright - 2017
+* This software is copyrighted to contributors listed in CONTRIBUTIONS.md.
+*
+* SPDX-License-Identifier: MIT
+*
+* Author(s) of this file:
+* J.Harding
+*
+* This is the base controller for the components to hold some commone methods.
+*/
+
+/**
+ * Tracks constants for chart names.
+ */
+const chartNameEnum = {
+  dateChart: 'date' // Track date as a chart to tell what metrics are dates
+}
+
+export default class BaseStatsComponent {
+
+    constructor ( $http, $filter ) {
+        Object.assign( this, { $http, $filter} )
```

Comment:
2 spaces is what I use and most JS developers use. I don't know if it's 'standard' across everyone though. I use this style guide. https://github.com/toddmotto/angular-styleguide